src/libraries/System.Drawing.Common/src/Interop/Windows/Interop.Gdi32.cs

```diff
@@ -188,29 +188,35 @@ internal sealed class DOCINFO
             internal int fwType;
 
 #if NET7_0_OR_GREATER
-            [CustomTypeMarshaller(typeof(DOCINFO), Direction = CustomTypeMarshallerDirection.In, Features = CustomTypeMarshallerFeatures.UnmanagedResources)]
-            internal struct Native
+            [ManagedToUnmanagedMarshallers(typeof(DOCINFO), InMarshaller = typeof(Marshaller))]
+            public static class Marshaller
             {
-                internal int cbSize;
-                internal IntPtr lpszDocName;
-                internal IntPtr lpszOutput;
-                internal IntPtr lpszDatatype;
-                internal int fwType;
+                public static Native ConvertToUnmanaged(DOCINFO managed) => new(managed);
+                public static void Free(Native native) => native.FreeNative();
```

Comment:
Should we be concerned with the size of the copy in this case? This doesn't seem too bad, but if a native type is large enough, no?

Reply:
Is a `Free(ref Native native)` better?

Reply:
That's an interesting point and definitely worth considering. Maybe we can optionally allow this to be a `ref` parameter (I don't know if I'd want to require it to be a `ref`, but we could also require it).

Reply:
We can talk about this in API review on Tuesday